File: second_homework/shapes/parallelepiped.cpp
#include "iostream"

#include "parallelepiped.h"

parallelepiped::parallelepiped(int width, int length, int height, double density) : shape(density) {
    this->width = width;
    this->length = length;
    this->height = height;
}

double parallelepiped::calculate_surface_area() {
    return 2.0 * (this->height * this->height + this->width * this->length + this->width * this->height);
}

void parallelepiped::to_string(FILE *outputFile) {
    fprintf(outputFile, "[PARALLELEPIPED] Width = %d Length = %d Height = %d Surface Area = %lf Density = %lf\n",
            width, length, height, calculate_surface_area(), density);
}

File: third_homework/shapes/parallelepiped.py
from shapes import shape


class Parallelepiped(shape.Shape):
    def __init__(self, width, length, height, density):
        super(Parallelepiped, self).__init__(density)
        self.width = width
        self.length = length
        self.height = height

    @property
    def surface_area(self):
        return 2 * (self.height * self.length + self.width * self.length + self.width * self.height)

    def __str__(self):
        return f'[PARALLELEPIPED] width = {self.width} | length = {self.length} | ' \
               f'height = {self.height} | density = {self.density} | surface_area = {self.surface_area}'
